lib/sorbet-rails/model_plugins/active_record_assoc.rb

```diff
@@ -137,14 +137,18 @@ def populate_collection_assoc_getter_setter(assoc_module_rbi, assoc_name, reflec
       assoc_name.to_s,
       return_type: relation_class,
     )
-    unless assoc_should_be_untyped?(reflection)
+    id_column = reflection.klass.primary_key
+    # For DB views, the PK column would not exist.
+    id_column = reflection.klass.columns_hash[id_column] if id_column
+    if id_column && !assoc_should_be_untyped?(reflection)
```

Comment:
it probably needs to check `reflection` first before retrieving value from `reflection.klass.primary_key`